kolibri/core/content/management/commands/importcontent.py

```diff
@@ -408,13 +420,10 @@ def _start_file_transfer(self, f, filetransfer, overall_progress_update):
             * FILE_TRANSFERRED - successfully transfer the file.
             * FILE_SKIPPED - the file does not exist so it is skipped.
         """
-        with filetransfer, self.start_progress(
-            total=filetransfer.total_size
-        ) as file_dl_progress_update:
+        with filetransfer:
             for chunk in filetransfer:
                 length = len(chunk)
                 overall_progress_update(length)
```

Comment:
I assume this is working because we are passing this in as an argument to the function, and so it's referring out to the thread local object in the parent thread?

Reply:
I'm not familiar with what part(s) specifically of the progress updating are not thread safe, but I have not yet seen any issues with it, and most of the code I've seen is using Python primitives, which are mostly thread-safe. If there are some specific things you'd like me to look into more, just let me know what parts you'd like me to check!

Reply:
The possible issue arises from the fact that the `get_task_object` helper function is thread safe - so if that's being invoked inside the thread then it would be an issue. Fortunately, we seem to just run this once in the main thread of the job: https://github.com/learningequality/kolibri/blob/release-v0.14.x/kolibri/core/tasks/management/commands/base.py#L112 so don't think there's any issues here!